### ISSUE TITLE
Slowly training speed?

### ISSUE THREAD

**Issue by JerryLeolfl** [junyanz/pytorch-CycleGAN-and-pix2pix#773]
Thanks for your great work in Cycle-GAN. However, when I use it in my data training, the training process seems so slow, almost 7 epochs a day. My training data contains 22000 images with size 256x256 in trainA. And the loss-curves seem no obviously changes.

**Comment by junyanz** [comment]
It is a little bit slow. Which GPU are you using? How fast is the data loading time (you can read it from the logging information). Consider using multiple faster GPUs or increase `--num_threads` (if data loading is the issue).

**Comment by JerryLeolfl** [comment]
Thanks a lot for your reply. The logging information is
epoch: 19, iters: 11932, time: 0.347, data: 0.001
epoch: 19, iters: 12032, time: 0.533, data: 0.001
...
I use 1080Ti GPU for training. And I tried to set _--gpu_ids '0,1'_. But it seems it still run only on gpu '0'. I cannot figure out the reason.
The loss curve is shown as follow:
![loss](https://user-images.githubusercontent.com/34597963/65366612-de84f900-dc58-11e9-96c0-4e6c17829736.png)
It seems the curve has no obviously changes.

**Comment by JerryLeolfl** [comment]
The loss hasn't converged, but the quality of output image is getting rising while training.

**Comment by junyanz** [comment]
Your running time looks normal. Your loss curves look fine. The LSGAN/DCGAN losses often oscillate and will not converge.  To use multiple GPUs, you need to increase the batch size `--batch_size`.
As you have 22000 images, you probably only need 20 to 50 epochs.

**Comment by JerryLeolfl** [comment]
Thanks for your answer. But I wonder without the loss curve as indicator, what can I refer to stop the training?

**Comment by junyanz** [comment]
GANs evaluation is an open problem. You can either evaluate your model on your downstream task or application (recommended) or using some standard metrics like FID.

**Comment by JerryLeolfl** [comment]
Thank you very much for your professional answer. I think my problem has been solved with your help.
